cast/src/lib.rs

```diff
@@ -146,6 +146,31 @@ where
         Ok::<_, eyre::Error>(res)
     }
 
+    /// Publishes a raw transaction to the network
+    ///
+    /// ```no_run
+    /// use cast::Cast;
+    /// use ethers_providers::{Provider, Http};
+    ///
+    /// # async fn foo() -> eyre::Result<()> {
+    /// let provider = Provider::<Http>::try_from("http://localhost:8545")?;
+    /// let cast = Cast::new(provider);
+    /// let res = cast.publish("0x1234".to_string()).await?;
+    /// println!("{:?}", res);
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn publish(&self, raw_tx: String) -> Result<PendingTransaction<'_, M::Provider>> {
+        let edit_raw_tx: String = match raw_tx.starts_with("0x") {
+            true => raw_tx.split_at(2).1.to_string(),
+            false => raw_tx,
+        };
+        let tx = Bytes::from(hex::decode(edit_raw_tx)?);
```

Comment:
I think this is more idiomatic:

```suggestion
    pub async fn publish(&self, mut raw_tx: String) -> Result<PendingTransaction<'_, M::Provider>> {
        raw_tx = match raw_tx.strip_prefix("0x") {
            Some(s) => s.to_string(),
            None => raw_tx,
        };
        let tx = Bytes::from(hex::decode(raw_tx)?);
```

Reply:
Or even
```suggestion
    pub async fn publish(&self, raw_tx: String) -> Result<PendingTransaction<'_, M::Provider>> {
        let tx = match raw_tx.strip_prefix("0x") {
            Some(raw) => Bytes::from(hex::decode(raw)?),
            None => Bytes::from(hex::decode(raw_tx)?),
        };
```

Reply:
thank you. I am still a noobie with rust so this is very helpful.